keyserver/handler.go

```diff
@@ -19,10 +19,22 @@ func (server *ConiksServer) handleRegistrationMessage(reg *p.RegistrationRequest
 			p.Error(p.ErrorMalformedClientMessage)
 	}
 
+	server.Lock()
+	// check the temporary bindings array first
+	// currently the server allows only one registration/key change per epoch
+	if server.tbs[reg.Username] != nil {
```

Comment:
If so, I would return the proof of inclusion along with the error if the registering name also already exists in the tree.